code/game/objects/effects/decals/Cleanable/fuel.dm

```diff
@@ -1,57 +1,66 @@
-obj/effect/decal/cleanable/liquid_fuel
+/obj/effect/decal/cleanable/liquid_fuel
 	//Liquid fuel is used for things that used to rely on volatile fuels or plasma being contained to a couple tiles.
 	icon = 'icons/effects/effects.dmi'
 	icon_state = "fuel"
 	layer = TURF_LAYER+0.2
 	anchored = 1
-	var/amount = 1 //Basically moles.
+	var/amount = 1 //Basically moles
+	var/newprocess = 1
+	var/processing = 0
 
-	New(newLoc,amt=1)
-		src.amount = amt
 
-		//Be absorbed by any other liquid fuel in the tile.
-		for(var/obj/effect/decal/cleanable/liquid_fuel/other in newLoc)
-			if(other != src)
-				other.amount += src.amount
-				spawn other.Spread()
-				qdel(src)
+/obj/effect/decal/cleanable/liquid_fuel/New(newLoc,amt=1)
+	src.amount = amt
 
-		Spread()
-		. = ..()
+	//Be absorbed by any other liquid fuel in the tile.
+	for(var/obj/effect/decal/cleanable/liquid_fuel/other in newLoc)
+		if(other != src)
+			other.amount += src.amount
+			spawn other.Spread()
+			qdel(src)
 
-	proc/Spread()
-		//Allows liquid fuels to sometimes flow into other tiles.
-		if(amount < 0.5) return
-		var/turf/simulated/S = loc
-		if(!istype(S)) return
-		for(var/d in cardinal)
-			if(rand(25))
-				var/turf/simulated/target = get_step(src,d)
-				var/turf/simulated/origin = get_turf(src)
-				if(origin.CanPass(null, target, 0, 0) && target.CanPass(null, origin, 0, 0))
-					if(!locate(/obj/effect/decal/cleanable/liquid_fuel) in target)
-						new/obj/effect/decal/cleanable/liquid_fuel(target, amount*0.25)
-						amount *= 0.75
+	Spread()
+	. = ..()
 
-	flamethrower_fuel
-		icon_state = "mustard"
-		anchored = 0
-		New(newLoc, amt = 1, d = 0)
-			dir = d //Setting this direction means you won't get torched by your own flamethrower.
-			. = ..()
+/obj/effect/decal/cleanable/liquid_fuel/proc/Spread()
+	//Allows liquid fuels to sometimes flow into other tiles.
+	if(amount < 0.5) return
+	var/turf/simulated/S = loc
+	if(!istype(S)) return
+	for(var/d in cardinal)
+		if(rand(25))
+			var/turf/simulated/target = get_step(src,d)
+			var/turf/simulated/origin = get_turf(src)
+			if(origin.CanPass(null, target, 0, 0) && target.CanPass(null, origin, 0, 0))
+				if(!locate(/obj/effect/decal/cleanable/liquid_fuel) in target)
+					new/obj/effect/decal/cleanable/liquid_fuel(target, amount*0.25)
+					amount *= 0.75
 
-		Spread()
-			//The spread for flamethrower fuel is much more precise, to create a wide fire pattern.
-			if(amount < 0.1) return
-			var/turf/simulated/S = loc
-			if(!istype(S)) return
+/obj/effect/decal/cleanable/liquid_fuel/fire_act(null, temperature, volume)
+	if(processing)
+		return
+	processing_objects.Add(src)
+	process()
```

Comment:
This shouldn't be directly called, processing_objects will call it on the next cycle.